### DISCUSSION TITLE
Setting `localContext` for all uses of some Type

### DISCUSSION THREAD

**Discussion by andy-donzelli-privitar** [graphql-java/graphql-java#2838] [Q&A] [answered]
I'm trying to use `DataFetchingEnvironment.getLocalContext()` to pass information from a root DTO to a child DTO. The use case is very similar to the one explained in your blog post [Building efficient data fetchers by looking ahead](https://www.graphql-java.com/blog/deep-dive-data-fetcher-results/).

How would that solution change if the `Issues` GQL Type appeared in the responses of a large number of Queries? I would like to map a single `DataFetcher` for all fields of GQL Type `Issues`. So far, the only solution to this I've come up with is to add a new `TypeWiring` for every GQL Type that includes the `Issues` Type in one of its Fields.

So, for example, if we had this schema:
```graphql
type Report {
   name: String!
   issues: Issues
}

type Epic {
   name: String!
   issues: Issues
}
```

then our Runtime Wiring would need
```java
.type(TypeRuntimeWiring.newTypeWiring("Report").dataFetcher("issues", issuesDataFetcher()))
.type(TypeRuntimeWiring.newTypeWiring("Epic").dataFetcher("issues", issuesDataFetcher()))
```

This is example is a bit contrived. Obviously two DataFetcher definitions aren't many, but you could imagine how this would become more troublesome when applied to a commonly used  GQL Type such as "User" or "Tag" that might appear in tens of other Types.

Does `graphql-java` support specifying a DataFetcher for all uses of a given Type?

**Comment by bbakerman** [comment, marked as answer]
> Does graphql-java support specifying a DataFetcher for all uses of a given Type?

Have a look at `graphql.schema.idl.WiringFactory#providesDataFetcher` and `graphql.schema.idl.WiringFactory#getDataFetcher`

This is a dynamic way to inject in data fetchers - above you used a static mechanism

`graphql.schema.idl.FieldWiringEnvironment` which is passed in contains information about a field including its "Type" so you could re-use the same fetcher say if it suits you
